velox/functions/lib/LikeFunctionUtils.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <velox/type/StringView.h>
+#include <velox/vector/TypeAliases.h>
+#include <string>
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+// This function checks if the string 'pattern' has wildcard characters ('_' or
+// '%'). It returns true if wildcard characters are found, else false.
+bool isWildcardInPattern(const StringView& pattern);
```

Comment:
StringView object can be passed by value.

Reply:
Could you please share if it should it be passed by value everywhere or just here?

Reply:
StringViews and std::string_views can be passed by value everywhere. These are light-weight objects.